src/app/qgsmaptooladdfeature.cpp

```diff
@@ -91,7 +92,7 @@ void QgsMapToolAddFeature::digitized( const QgsFeature &f )
     if ( topologicalEditing )
     {
       QList<QgsPointLocator::Match> sm = snappingMatches();
-      Q_ASSERT( f.geometry().constGet()->vertexCount() == sm.size() );
+      Q_ASSERT( f.geometry().constGet()->vertexCount() == ( vlayer->geometryType() == QgsWkbTypes::PolygonGeometry ? sm.size() + 1 : sm.size() ) );
```

Comment:
Could the same also happen on a ring line geometry? Maybe just a `vertexCount > sm.size()`?
Disclosure: I don't have the complete picture.

Reply:
@m-kuhn , define a ring line geometry? To my knowledge, only polygons end up with vertices entered + 1 when its geometry is created.

Reply:
It's also possible to draw a line that has multiple vertices on top of each other. I wonder if snapping kills such duplicates. If not, please ignore my comment.

Reply:
@m-kuhn , let's go with this for now -- if I stumble on more bogus assert here, I'll fix along the road. I.e., I'm also missing a bigger picture at this moment in time, therefore I'm inclined to stick as closely as possible to the original assert's intent.